### PR TITLE
Feature/overlay

### PR DESCRIPTION
**Pull request by wjian23**

### CODE REVIEW

File: core/src/widget.rs [hunk 1]
```diff
@@ -78,6 +78,57 @@ pub enum Widget {
...
+}
+
+pub struct ClosureWidget<F, R, M>
+where
+  F: FnOnce(&BuildCtx) -> R + 'static,
+  R: IntoWidget<M>,
+  M: ImplMarker,
+{
+  closure: F,
+  marker: PhantomData<M>,
```
Comment (M-Adoo) [inline core/src/widget.rs:92]: unnecesseary marker

File: core/src/overlay.rs [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+
+#[derive(Declare)]
+pub struct Overlay<M: ImplMarker, W: IntoWidget<M> + Clone> {
+  content: W,
+  #[declare(default=Some(ModalConfig::default()), convert=strip_option)]
+  modal: Option<ModalConfig>,
+  #[declare(skip)]
+  handle: Option<OverlayHandle>,
+  #[declare(skip)]
+  _marker: PhantomData<M>,
```
Comment (M-Adoo) [inline core/src/overlay.rs:48]: unnecessary marker

File: core/src/overlay.rs [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+    let Some(modal) = &self.modal else { return w };
+    w = widget! {
+      Container {
+        size: Size::new(f32::INFINITY, f32::INFINITY),
+        Widget::new(w)
+      }
+    }
+    .into_widget();
+
+    if modal.close_on.contains(ClosePolicy::CLOSE_ON_ESCAPE) {
```
Comment (M-Adoo) [inline core/src/overlay.rs:170]: Declare with `DynWidget` more clear, and have same result widget tree.

File: widgets/src/layout/flex.rs [hunk 4]
```diff
@@ -208,7 +208,7 @@ impl FlexLayouter {
           line.main_width += gap;
         }
 
-        let mut line = &mut self.current_line;
+        let line = &mut self.current_line;
```
Comment (sologeek) [inline widgets/src/layout/flex.rs:211]: Why can remove `mut` here? the next line change it's field.

**Pull request closed without merging**